inductiva/tasks/task.py

```diff
@@ -0,0 +1,147 @@
+"""Interact with a running/completed task on the Inductiva API."""
+import pathlib
+import time
+from typing import Dict, Any
+from typing_extensions import TypedDict
+
+from inductiva.client.models import TaskStatusCode
+from inductiva import api
+from inductiva.client.apis.tags.tasks_api import TasksApi
+from absl import logging
+
+
+class Task:
+    """Represents a running/completed task on the Inductiva API.
+
+    Attributes:
+        id: The task ID.
+        _api: Instance of TasksApi (from the generated API client).
+    """
+
+    def __init__(self, task_id: str):
+        """Initialize the instance from a task ID."""
+        self.id = task_id
+        self._api = TasksApi(api.get_client())
+
+    def __enter__(self):
+        """Enter context manager for managing a blocking execution.
+
+        If an exception/ctrl+c is caught while in the context manager, the task
+        is killed.
+
+        Usage:
+            task = scenario.simulate_async(...)
+            with task:
+                # If an exception happens here or ctrl+c is pressed, the task
+                # will be killed.
+                task.wait()
+
+        """
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """Exit context manager killing the task if an exception was raised."""
+        del traceback  # unused
+        del exc_type  # unused
+
+        if exc_value is None:
+            return True
+
+        if isinstance(exc_value, KeyboardInterrupt):
+            logging.info("Caught SIGINT: terminating blocking task...")
+        elif exc_value is not None:
+            logging.info("Caught exception: terminating blocking task...")
+
+        self.kill()
+        return False
+
+    def get_status(self) -> TaskStatusCode:
+        """Get status of the task.
+
+        This method issues a request to the API.
+        """
+        resp = self._api.get_task_status(self._get_path_params())
+        return TaskStatusCode(resp.body["status"])
+
+    def get_info(self) -> Dict[str, Any]:
+        """Get a dictionary with information about the task.
+
+        This method issues a request to the API.
+        """
+        params = self._get_path_params()
+        resp = self._api.get_task(params, skip_deserialization=True).response
+        return resp.json()
+
+    def wait(self, polling_period: int = 5) -> TaskStatusCode:
```

Comment:
This is great and seems to provide both functionality to manage the tasks in the backend and frontend.